libs/feature/teaching/src/lib/course/course-learnpath-editor/learnpath-editor.tsx

```diff
@@ -0,0 +1,311 @@
+			const mesh: Mesh = {
+				requiredSkills: convertToMeshSkills(lesson.requirements),
+				lesson: {
+					lessonId: lesson.lessonId,
+					title: lesson.title
+				},
+				gainedSkills: convertToMeshSkills(lesson.teachingGoals)
+			};
+			meshes.push(mesh);
+		}
+	}
+
+	return meshes;
+}
+
+function isLessonInMeshes(mesh: Mesh, meshes: Mesh[]) {
+	const temp = meshes.filter(elemt => elemt.lesson.title === mesh.lesson.title);
+	if (temp.length > 0) {
+		return true;
+	} else {
+		return false;
+	}
+}
+
+export function LearnpathEditor({
+	repositoryId,
+	teachingGoals,
+	lessonPoolId
+}: {
+	repositoryId: string;
+	teachingGoals: SkillFormModel[];
+	lessonPoolId: string;
+}) {
+	const initMeshes: Mesh[] = createMeshes(lessonPoolId);
+	const initMesh: Mesh = {
+		requiredSkills: [],
+		lesson: { title: "", lessonId: "" },
+		gainedSkills: []
+	};
+
+	const [meshes, setMeshes] = useState(initMeshes);
+	const [displayedMesh, setDisplayedMesh] = useState(initMesh);
+	const [isSidebarOpen, setIsSidebarOpen] = useState(false);
+
+	const graph = convertToGraph(meshes, false);
+	const learnpath = generateLearnpathFromMeshes(teachingGoals, repositoryId, meshes);
+
+	const { mutateAsync: updateLessonsInPool } = trpc.lessonPool.updateLessons.useMutation();
+	const { mutateAsync: updateLessonSkills } = trpc.lesson.updateSkills.useMutation();
+
+	const addMesh = async (mesh: Mesh) => {
+		const updatedMeshes = [...meshes];
+		if (!isLessonInMeshes(mesh, meshes)) {
+			updatedMeshes.push(mesh);
+		} else {
+			const meshToEdit = meshes.filter(elem => elem.lesson.title === mesh.lesson.title)[0];
+			meshToEdit.requiredSkills = mesh.requiredSkills;
+			meshToEdit.gainedSkills = mesh.gainedSkills;
+			updatedMeshes.map(elem => elem.lesson.title === mesh.lesson.title || meshToEdit);
+		}
+		setMeshes(updatedMeshes);
+		if (mesh.lesson.lessonId !== null) {
+			try {
+				const result = await updateLessonSkills({
+					id: mesh.lesson.lessonId,
+					requirements: mesh.requiredSkills,
+					teachingGoals: mesh.gainedSkills
+				});
+				console.log("Updating lesson", result);
+			} catch (error) {
+				console.log("Error while updating lesson", error);
+			}
+		}
+	};
+
+	const getLessonIds = () => {
+		const lessonsIds = meshes.map(mesh => mesh.lesson.lessonId);
+		const lessonIds = lessonsIds.filter(elem => elem !== null);
+		return lessonIds.map(elem => elem as string);
+	};
+
+	const updateLessonPool = async () => {
+		const lessonIds: string[] = getLessonIds();
+		try {
+			const response = await updateLessonsInPool({
+				id: lessonPoolId,
+				lessons: lessonIds
+			});
+			console.log("Updating lesson pool successful", response);
+		} catch (error) {
+			console.log(error);
+		}
+	};
+
+	// TODO
+	const loadLessonPool = () => {
+		setMeshes(initMeshes);
+	};
+
+	const removeMesh = (meshToRemove: Mesh) => {
+		const updatedMeshes = meshes.filter(
+			mesh => mesh.lesson.title !== meshToRemove.lesson.title
+		);
+		setMeshes(updatedMeshes);
```

Comment:
hier wichtig den alternativen setter zu verwenden. 
```
setMeshes(previousMesh => previousMesh.filter(m => m.lesson.title !== meshToRemove.lesson.title)
```
das ist insbesondere für state batching wichtig https://react.dev/learn/queueing-a-series-of-state-updates